crates/partition-store/src/partition_store.rs

```diff
@@ -465,37 +464,34 @@ impl PartitionStore {
         snapshot_base_path: &Path,
         min_target_lsn: Option<Lsn>,
         snapshot_id: SnapshotId,
-    ) -> std::result::Result<LocalPartitionSnapshot, SnapshotError> {
+    ) -> Result<LocalPartitionSnapshot> {
         let applied_lsn = self
             .get_applied_lsn()
             .await
-            .map_err(|e| SnapshotError::SnapshotExport(self.partition_id, anyhow!(e)))?
-            .ok_or(SnapshotError::SnapshotExport(
-                self.partition_id,
-                anyhow!(StorageError::DataIntegrityError),
-            ))?;
+            .map_err(|e| StorageError::SnapshotExport(e.into()))?
+            .ok_or(StorageError::DataIntegrityError)?; // if PartitionStore::get_applied_lsn returns None
 
         if let Some(min_target_lsn) = min_target_lsn {
             if applied_lsn < min_target_lsn {
-                return Err(SnapshotError::MinimumTargetLsnNotMet {
-                    partition_id: self.partition_id,
-                    min_target_lsn,
+                return Err(StorageError::PreconditionFailed(anyhow!(
+                    "Applied LSN below the target LSN: {} < {}",
                     applied_lsn,
-                });
+                    min_target_lsn
+                )));
             }
         };
 
         // RocksDB will create the snapshot directory but the parent must exist first:
         tokio::fs::create_dir_all(snapshot_base_path)
             .await
-            .map_err(|e| SnapshotError::SnapshotIo(self.partition_id, e))?;
+            .map_err(|e| StorageError::SnapshotExport(e.into()))?;
```

Comment:
I see this pattern a lot. While it's correct I usually don't like repeating code :) Instead I would suggest to do something like

```rust
impl StorageError {
    pub fn snapshot_export<E: Into<anyhow::Error>>(err: E) -> Self {
        Self::SnapshotExport(err.into())
    }
}
```

Then in places where it's used like above it can be like 
```suggestion
            .map_err(StorageError::snapshot_export)?;
```

This is just to make it look better, there is absolutely nothing wrong with what you have.

This can also be done for `SnapshotError`